src/StoreApi/Authentication.php

```diff
@@ -26,6 +26,11 @@ public function check_authentication( $result ) {
 			return $result;
 		}
 
+		// Disable Rate Limiting for logged in users with 'edit posts' capability.
+		if ( current_user_can( 'edit_posts' ) ) {
```

Comment:
@tarhi-saad I wonder if at this point, we should move Rate Limiting logic to its own method (it kind of falls outside the naming scope of this one) and avoid the double return of the ternary. Something along the lines of:

```
if ( ! current_user_can( 'edit_posts' ) ) {
    $this->apply_rate_limiting();
}
```

Reply:
> I wonder if at this point, we should move Rate Limiting logic to its own method

I've created a separate function for it in [this commit](https://github.com/woocommerce/woocommerce-blocks/pull/7934/commits/d608e1c39cb08c9c21ff813b7b83dded9d77eddb). But, for some reason, the rate-limiting wasn't working anymore. So, I reverted the change! Maybe I made a mistake in my refactoring!